### DISCUSSION TITLE
Unable to draw rect at 0,0 when window maximised on MacOS

### DISCUSSION THREAD

**Discussion by J7mbo** [hajimehoshi/ebiten#2135] [Q&A] [answered]
Hi! I'm new to ebiten. I'm trying to draw an image at 0,0. This works fine when the window is like it is in the examples, however when the window is maximised and I draw at (what I think should be) 0,0, there's a large margin on the left.

Windowed, works fine:
<img width="109" alt="image" src="https://user-images.githubusercontent.com/2657310/173097351-d384a5b8-f883-44cf-abcb-a6858012ce37.png">

With `ebiten.SetFullscreen(true)`:

<img width="230" alt="image" src="https://user-images.githubusercontent.com/2657310/173097673-5800a6f7-ba0c-4d06-bc5f-c0df22b9df44.png">

_Please note that I'm moving the mouse to the top of the screen here to show you where the box starts_

Here's the ebiten code:

```
    ebiten.SetWindowSize(800, 600)
    ebiten.SetWindowDecorated(true)
    ebiten.SetWindowResizingMode(ebiten.WindowResizingModeEnabled)
    ebiten.SetFullscreen(true)
```

And the draw rect code (I've used multiple libraries and I don't believe it's related, could do it by hand too) :

```
import "github.com/fogleman/gg"

func init() {
        dc := gg.NewContext(800, 600)
	dc.DrawRectangle(0, 0, 50, 50)
	dc.SetColor(color.White)
	dc.Fill()

        // Image is the *ebiten.Image from the tutorial
	img = ebiten.NewImageFromImage(dc.Image())
}

func (g *Game) Draw(screen *ebiten.Image) {
	screen.DrawImage(img, nil)
}
```

Why can't I display a rectangle at 0,0 when full-screened? Am I doing something wrong here?

**Comment by hajimehoshi** [comment, marked as answer]
It's because the aspect ratios (the actual screen size and the returning value of `Layout`) don't match. What is the implementation of `Layout`? Thanks,

**Reply by J7mbo** [reply]
Thanks for such a fast response @hajimehoshi! Sat here scratching my head for a day already...

```
func (g *Game) Layout(outsideWidth, outsideHeight int) (screenWidth, screenHeight int) {
	return 800, 600
}
```

So the `Layout` is definitely the same as what I pass in `SetWindowSize`.

**Reply by hajimehoshi** [reply]
When your game is in the fullscreen mode, the outside size's aspect ratio is different from the aspect ratio of 800x600. If you want to adjust the game screen size with the outside size,

```go
func (g *Game) Layout(outsideWidth, outsideHeight int) (screenWidth, screenHeight int) {
	return outsideWidth, outsideHeight
}
```

should work.

**Reply by J7mbo** [reply]
Additional info: When I maximise the window myself, there is a second where the white box is indeed in the top left corner at 0,0, but with the wrong aspect ratio. But then the box jumps over to the right as in the second screenshot, and the aspect ratio is fixed.

**Reply by J7mbo** [reply]
Ah yes, your change to `return` fixed this. Facepalm here. Many thanks hajimehoshi! Wishing you a great weekend :)

**Reply by hajimehoshi** [reply]
I'm very glad that your issue was solved. Thank you for using Ebitengine!
